### ISSUE TITLE
Add CollectionContainsAll Predicate

### ISSUE THREAD

**Issue by d47853** [gchq/koryphe#211]
Add a Predicate which stores a number of values and ensures a collection contains all the values.

**Comment by n3101** [comment]
@d47853 Is this different to https://gchq.github.io/gaffer-doc/getting-started/predicates/arein.html

**Comment by d47853** [comment]
Yes subtly. It's flipped. AreIn is checking if the values in the input all exist in a predefined set. Whereas the CollectionContainsAll would check that values in the input contain all the values in your predefined set.
